FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,59 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		performAction { context in
+			do {
+				let fetchRequest = NSFetchRequest<ManagedCache>(entityName: ManagedCache.className())
+				if let managedCache = try context.fetch(fetchRequest).first {
+					let managedFeed = managedCache.feed.compactMap { $0 as? ManagedFeedImage }.map {
+						LocalFeedImage(id: $0.id, description: $0.imageDescription, location: $0.location, url: $0.url)
+					}
+					completion(.found(feed: managedFeed, timestamp: managedCache.timestamp))
+				} else {
+					completion(.empty)
+				}
```

Comment:
This code duplicates `ManagedCache.find(in:)` and thus can be simplified to the following
```suggestion
				if let managedCache = try ManagedCache.find(in: context) {
					let managedFeed = managedCache.feed.compactMap { $0 as? ManagedFeedImage }.map {
						LocalFeedImage(id: $0.id, description: $0.imageDescription, location: $0.location, url: $0.url)
					}
					completion(.found(feed: managedFeed, timestamp: managedCache.timestamp))
				} else {
					completion(.empty)
				}
```

Reply:
Done.